### PR TITLE
BPU Prep Work Part2

### PR DESCRIPTION
**Pull request by Kukks**
* Adjust tests to use the hot wallet when registering deriv scheme
* Add amount to payment data view for onchain payments
* Make zone limits higher when in dev mode (for tests in next PR)
* Make IPaymentMethodDetails serialize/deserialize through payment type using the network
* Allow named settings through settings repo
* Refactor some extensions for next PR

### CODE REVIEW

File: BTCPayServer/Services/Invoices/InvoiceEntity.cs [hunk 1]
```diff
@@ -821,8 +821,7 @@ public PaymentMethod SetPaymentMethodDetails(IPaymentMethodDetails paymentMethod
                 FeeRate = bitcoinPaymentMethod.FeeRate;
                 DepositAddress = bitcoinPaymentMethod.DepositAddress;
             }
-            var jobj = JObject.Parse(JsonConvert.SerializeObject(paymentMethod));
-            PaymentMethodDetails = jobj;
+            PaymentMethodDetails = JObject.Parse(paymentMethod.GetPaymentType().SerializePaymentMethodDetails(Network, paymentMethod));;
```
Comment (NicolasDorier) [inline BTCPayServer/Services/Invoices/InvoiceEntity.cs]: remove dups `;;`

File: BTCPayServer/Hosting/BTCPayServerServices.cs [hunk 2]
```diff
@@ -260,10 +260,20 @@ public static IServiceCollection AddBTCPayServer(this IServiceCollection service
...
-            var rateLimits = new RateLimitService();
-            rateLimits.SetZone($"zone={ZoneLimits.Login} rate=5r/min burst=3 nodelay");
-            services.AddSingleton(rateLimits);
+            services.AddSingleton(provider =>
+            {
+                var btcPayEnv = provider.GetService<BTCPayServerEnvironment>();
+                var rateLimits = new RateLimitService();
+                if (btcPayEnv.IsDevelopping)
+                {
+                    rateLimits.SetZone($"zone={ZoneLimits.Login} rate=1000r/min burst=100 nodelay");
```
Comment (NicolasDorier) [inline BTCPayServer/Hosting/BTCPayServerServices.cs]: wtf is the problem with you development machine to reach this issue??

File: BTCPayServer/Extensions.cs [hunk 3]
```diff
@@ -257,6 +265,12 @@ public static bool IsOnion(this HttpRequest request)
                 return false;
             return request.Host.Host.EndsWith(".onion", StringComparison.OrdinalIgnoreCase);
         }
+        
+        public static bool IsOnion(this Uri uri)
+        {
+            return uri?.DnsSafeHost?.EndsWith(".onion", StringComparison.OrdinalIgnoreCase) ?? false;
```
Comment (NicolasDorier) [inline BTCPayServer/Extensions.cs]: `return uri?.DnsSafeHost?.EndsWith(".onion", StringComparison.OrdinalIgnoreCase) is true;`

File: BTCPayServer/Hosting/BTCPayServerServices.cs (hunk 2 above)
Comment (Kukks) [reply BTCPayServer/Hosting/BTCPayServerServices.cs]: I have a test that does a bunch of calls in sequence which triggered the default rate limiter.

File: BTCPayServer/Payments/PaymentTypes.Bitcoin.cs [hunk 4]
```diff
@@ -31,9 +31,14 @@ public override string SerializePaymentData(BTCPayNetworkBase network, CryptoPay
             return ((BTCPayNetwork) network).ToString(paymentData);
         }
 
-        public override IPaymentMethodDetails DeserializePaymentMethodDetails(string str)
+        public override IPaymentMethodDetails DeserializePaymentMethodDetails(BTCPayNetworkBase network, string str)
```
Comment (NicolasDorier) [inline BTCPayServer/Payments/PaymentTypes.Bitcoin.cs]: Use the `JsonConvert.` for now.
It is not proven that using the network serializer will not fuck up existing objects by applying converters to the payment details.

**Pull request merged**